styx-cli/src/main/java/com/spotify/styx/cli/CliMain.java

```diff
@@ -380,6 +377,74 @@ private void workflowCreate() throws IOException, ExecutionException, Interrupte
     }
   }
 
+  private void workflowEnable() throws ExecutionException, InterruptedException {
+    final String component = namespace.getString(parser.workflowEnableComponentId.getDest());
+    final List<String> workflows = namespace.getList(parser.workflowEnableWorkflowId.getDest());
+    workflowStateUpdate(component, workflows, WorkflowState.patchEnabled(true))
+        .forEach(t -> {
+          if (t._2.enabled().orElse(false)) {
+            cliOutput.printMessage("Workflow " + t._1 + " in component " + component + " enabled.");
+          } else {
+            cliOutput.printMessage(
+                "Failed to enable workflow " + t._1 + " in component " + component + ".");
+          }
+        });
+  }
+
+  private void workflowDisable() throws ExecutionException, InterruptedException {
+    final String component = namespace.getString(parser.workflowDisableComponentId.getDest());
+    final List<String> workflows = namespace.getList(parser.workflowDisableWorkflowId.getDest());
+    workflowStateUpdate(component, workflows, WorkflowState.patchEnabled(false))
+        .forEach(t -> {
+          if (!t._2.enabled().orElse(false)) {
+            cliOutput.printMessage("Workflow " + t._1 + " in component " + component + " disabled.");
+          } else {
+            cliOutput.printMessage(
+                "Failed to disable workflow " + t._1 + " in component " + component + ".");
+          }
+        });
+  }
+
+  private List<Tuple2<String, WorkflowState>> workflowStateUpdate(String component,
+                                                                  List<String> workflows,
+                                                                  WorkflowState workflowState)
+      throws ExecutionException, InterruptedException {
+    final List<Tuple2<String, CompletionStage<WorkflowState>>> futures = workflows.stream()
+        .map(workflow -> Tuple.of(workflow,
+                                  styxClient.updateWorkflowState(component,
+                                                                 workflow,
+                                                                 workflowState)))
+        .collect(toList());
+
+    final List<Tuple2<String, WorkflowState>> updatedWorkflowStates =
```

Comment:
I guess this whole thing could be made into some kind of `map/flatMap` to wait for and process these futures but the checked exceptions make it quite ugly.

Reply:
Yeah, that's the main reason.